bootstrap-app/src/cmr/bootstrap/data/metadata_retrieval/collection_metadata_cache.clj

```diff
@@ -0,0 +1,123 @@
+                 * various format keys each mapped to compressed metadata."
+  (:require
+   [clj-time.core :as t]
+   [clojure.set :as cset]
+   [cmr.common-app.data.metadata-retrieval.collection-metadata-cache :as cmn-coll-metadata-cache]
+   [cmr.common-app.data.metadata-retrieval.revision-format-map :as crfm]
+   [cmr.bootstrap.data.metadata-retrieval.metadata-transformer :as metadata-transformer]
+   [cmr.common.config :refer [defconfig]]
+   [cmr.common.hash-cache :as hash-cache]
+   [cmr.common.jobs :refer [defjob]]
+   [cmr.common.log :as log :refer [info]]
+   [cmr.common.util :as u]
+   [cmr.metadata-db.services.concept-service :as metadata-db]
+   [cmr.umm-spec.versioning :as umm-version]))
+
+(defconfig non-cached-collection-metadata-formats
+  "Defines a set of collection metadata formats that will not be cached in memory"
+  {:default #{}
+   :type :edn})
+
+(def all-formats
+  "All the possible collection metadata formats that could be cached."
+  #{:echo10
+    :iso19115
+    :dif
+    :dif10
+    ;; Note that when upgrading umm version we should also cache the previous version of UMM.
+    {:format :umm-json
+     :version umm-version/current-collection-version}})
+
+(defn cached-formats
+  "This is a set of formats that are cached."
+  []
+  (cset/difference all-formats (non-cached-collection-metadata-formats)))
+
+(defn- concept-tuples->cache-map
+  "Takes a set of concept tuples fetches the concepts from metadata db, converts them to revision
+   format maps, and stores them into a cache map"
+  [context concept-tuples]
+  (let [mdb-context (cmn-coll-metadata-cache/context->metadata-db-context context)
+        concepts (doall (metadata-db/get-concepts mdb-context concept-tuples true))
+        concepts (cmn-coll-metadata-cache/concepts-without-xml-processing-inst concepts)
+        rfms (u/fast-map #(crfm/compress
+                           (crfm/concept->revision-format-map context 
+                                                              % 
+                                                              (cached-formats) 
+                                                              metadata-transformer/transform-to-multiple-formats 
+                                                              true))
+                         concepts)]
+    (reduce #(assoc %1 (:concept-id %2) %2) {} rfms)))
+
+(defn update-cache
+  "Updates the collection metadata cache by querying elastic search for updates since the 
+  last time the cache was updated."
+  [context]
+  (info "Updating collection metadata cache")
+  (let [incremental-since-refresh-date (str (t/now))
+        concepts-tuples (cmn-coll-metadata-cache/fetch-updated-collections-from-elastic context)
+        new-cache-value (reduce #(merge %1 (concept-tuples->cache-map context %2))
+                                {}
+                                (partition-all 1000 concepts-tuples))
```

Comment:
where does the 1000 value come from?

Reply:
Who ever wrote the code originally decided that splitting up the results into chunks of 1000 was the correct thing to do.  I just copied the original code into here.  It is possible that I could put this code into common, and pass in the metadata-transformer that is needed in the concept-tuples->cache-map.

Reply:
My only comment about that is that future developers don't know if 1000 is significant and if it can be changed. Not saying that you change it to :probable-chunk-size